### DISCUSSION TITLE
After updating to Stable v2021.09 salt-minion virt.py grain has syntax error

### DISCUSSION THREAD

**Discussion by beaconfield** [uyuni-project/uyuni#4277] [General]
I upgraded my Uyuni Server to 2021.09 today and then updated packages on a RHEL/OL 7 client. I noticed that there is still an error showing up for the salt-minion and virt.py grain. From my syslog:

Sep 30 14:27:38 <omitted> salt-minion[4488]: [ERROR   ] Failed to import grains virt, this is due most likely to a syntax error:
Sep 30 14:27:38 <omitted> salt-minion[4488]: Traceback (most recent call last):
Sep 30 14:27:38 <omitted> salt-minion[4488]: File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1658, in _load_module
Sep 30 14:27:38 <omitted> salt-minion[4488]: mod = imp.load_module(mod_namespace, fn_, fpath, desc)
Sep 30 14:27:38 <omitted> salt-minion[4488]: File "/var/cache/salt/minion/extmods/grains/virt.py", line 41
Sep 30 14:27:38 <omitted> salt-minion[4488]: matcher = re.search(rb'(\d+)\.(\d+)\.(\d+)', version_out)
Sep 30 14:27:38 <omitted> salt-minion[4488]: ^
Sep 30 14:27:38 <omitted> salt-minion[4488]: SyntaxError: invalid syntax

Should I report this as a bug? I cannot find anything related to it in the open issues list but I have not opened a bug on this project before so I want to make sure.

Thanks!

**Comment by agraul** [comment]
Welcome @beaconfield !

In general it is always a good idea to open an "Uyuni - Bugreport" type issue for tracebacks like the one you have. The template asks for infomation, like the traceback you've posted here, but if we need more information we will just ask you.

For this bug, the fix was already merged (https://github.com/uyuni-project/uyuni/commit/b8f7610a3cd05b651681e2d93ac9ec28f82a6e1b#diff-39c61a09d6a85fd93ff5d994d857b577bc87db925655afe783993e7d8d233309). The next release will include it. 

If you have any more questions, just ask away :) We're also in available in gitter or on the mailing lists.

**Reply by beaconfield** [reply]
> Welcome @beaconfield !
> 
> In general it is always a good idea to open an "Uyuni - Bugreport" type issue for tracebacks like the one you have. The template asks for infomation, like the traceback you've posted here, but if we need more information we will just ask you.
> 
> For this bug, the fix was already merged ([b8f7610#diff-39c61a09d6a85fd93ff5d994d857b577bc87db925655afe783993e7d8d233309](https://github.com/uyuni-project/uyuni/commit/b8f7610a3cd05b651681e2d93ac9ec28f82a6e1b#diff-39c61a09d6a85fd93ff5d994d857b577bc87db925655afe783993e7d8d233309)). The next release will include it.
> 
> If you have any more questions, just ask away :) We're also in available in gitter or on the mailing lists.

Thank you very much for the reply. That's very helpful to know. I was able to go into my Uyuni Server instance and fix that line (in the commit) on the virt.py grain and sure enough - when I restarted salt-minion on the client it refreshed the cache and it does not throw the error any longer. 

I'll look forward to testing it when the next Uyuni release is out. Thanks again!
